develop/dev-guide-sample-application-nodejs-mysql2.md

```diff
@@ -0,0 +1,311 @@
+
+1. Run the following command to copy `.env.example` and rename it to `.env`:
+
+    ```shell
+    cp .env.example .env
+    ```
+
+2. Edit the `.env` file, set up the environment variables as follows, replace the corresponding placeholders `{}` with connection parameters on the connection dialog:
+
+    ```dotenv
+    TIDB_HOST={host}
+    TIDB_PORT=4000
+    TIDB_USER=root
+    TIDB_PASSWORD={password}
+    TIDB_DATABASE=test
+    ```
+
+   If you are running TiDB locally, the default host address is `127.0.0.1`, and the password is empty.
+
+3. Save the `.env` file.
+
+</div>
+</SimpleTab>
+
+### Step 4: Run the code and check the result
+
+Run the following command to execute the sample code:
+
+```shell
+npm start
+```
+
+If the connection is successful, the console will output the version of the TiDB cluster as follows:
+
+```
+🔌 Connected to TiDB cluster! (TiDB version: 5.7.25-TiDB-v7.1.0)
+⏳ Loading sample game data...
+✅ Loaded sample game data.
+
+🆕 Created a new player with ID 12.
+ℹ️ Got Player 12: Player { id: 12, coins: 100, goods: 100 }
+🔢 Added 50 coins and 50 goods to player 12, updated 1 row.
+🚮 Deleted 1 player data.
+```
+
+## Sample code snippets
+
+You can refer to the following sample code snippets to complete your own application development.
+
+For complete sample code and how to run it, check out the [tidb-samples/tidb-nodejs-mysql2-quickstart](https://github.com/tidb-samples/tidb-nodejs-mysql2-quickstart) repository.
+
+### Connect with connection options
+
+The following code establish a connection to TiDB with options defined in the environment variables:
+
+```javascript
+// Step 1. Import the 'mysql' and 'dotenv' packages.
+import { createConnection } from "mysql2/promise";
+import dotenv from "dotenv";
+import * as fs from "fs";
+
+// Step 2. Load environment variables from .env file to process.env.
+dotenv.config();
+
+async function main() {
+   // Step 3. Create a connection with the TiDB cluster.
+   const options = {
+      host: process.env.TIDB_HOST || '127.0.0.1',
+      port: process.env.TIDB_PORT || 4000,
+      user: process.env.TIDB_USER || 'root',
+      password: process.env.TIDB_PASSWORD || '',
+      database: process.env.TIDB_DATABASE || 'test',
+      ssl: process.env.TIDB_ENABLE_SSL === 'true' ? {
+         minVersion: 'TLSv1.2',
+         ca: process.env.TIDB_CA_PATH ? fs.readFileSync(process.env.TIDB_CA_PATH) : undefined
+      } : null,
+   }
+   const conn = await createConnection(options);
+
+   // Step 4. Perform some SQL operations...
+
+   // Step 5. Close the connection.
+   await conn.end();
+}
+
+void main();
+```
+
+> **Note**
+>
+> For TiDB Serverless, TLS connection **MUST** be enabled via `TIDB_ENABLE_SSL` when using public endpoint, but you **don't** have to specify an SSL CA certificate via `TIDB_CA_PATH`, because Node.js uses the built-in [Mozilla CA certificate](https://wiki.mozilla.org/CA/Included_Certificates) by default, which is trusted by TiDB Serverless.
```

Comment:
```suggestion
> For TiDB Serverless, you **MUST** enable TLS connection via `TIDB_ENABLE_SSL` when using public endpoint. However, you **don't** have to specify an SSL CA certificate via `TIDB_CA_PATH`, because Node.js uses the built-in [Mozilla CA certificate](https://wiki.mozilla.org/CA/Included_Certificates) by default, which is trusted by TiDB Serverless.
```